src/main/resources/google_checks.xml

```diff
@@ -110,7 +110,7 @@
                value="COMMA, SEMI, TYPECAST, LITERAL_IF, LITERAL_ELSE, LITERAL_RETURN,
                     LITERAL_WHILE, LITERAL_DO, LITERAL_FOR, LITERAL_FINALLY, DO_WHILE, ELLIPSIS,
                     LITERAL_SWITCH, LITERAL_SYNCHRONIZED, LITERAL_TRY, LITERAL_CATCH, LAMBDA,
-                    LITERAL_YIELD, LITERAL_CASE"/>
+                    LITERAL_YIELD, LITERAL_CASE, LITERAL_WHEN"/>
```

Comment:
@mahfouz72 @nrmancuso @romani @Zopsss Same as other PR.
We need to show this follows google's style.

Reply:
my answer would be same as https://github.com/checkstyle/checkstyle/pull/15311#discussion_r1686824725. The style guide does not have rules for java 21 keywords but we're talking about basic spacing rules here, so I think we should add support for this token, it'll improve overall code readability.

Reply:
Please run Google formatter over Input to be double confident that space is required.
I think space should be there

Reply:
I have did it here: https://github.com/checkstyle/checkstyle/pull/15311#discussion_r1686824725

Formatter does adds space around `when`

Reply:
@mahfouz72 , please add few examples (copy from regular test Inputs is ok) of "when" to new Input at folder
https://github.com/checkstyle/checkstyle/tree/master/src/it/resources-noncompilable/com/google/checkstyle/test/chapter4formatting
and create test method at https://github.com/checkstyle/checkstyle/blob/master/src/it/java/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/HorizontalWhitespaceTest.java

this Check is in 4.6.2 https://checkstyle.org/google_style.html#a4.6.2

same request was done at https://github.com/checkstyle/checkstyle/pull/15311 , it might required to be added ones.

Reply:
https://github.com/checkstyle/checkstyle/pull/15359

Reply:
I am awaiting responses from @romani @nrmancuso on this. If this agreed to covered by the style guide, then we need to add examples to the IT inputs.

Reply:
@rnveach please see 2 comments above yours. I think @romani agrees and I already asked to extend IT inputs  there is a PR

Reply:
I agree to extend google style config.

Reply:
Ok, we will extend the google style config, I have removed the blocked label.

Reply:
ok. I don't understand why https://github.com/checkstyle/checkstyle/pull/15359 is a separate PR. It should be merged in with this since this is where it is being added to google config.

Reply:
> since this is where it is being added to google config.

but it is being added in https://github.com/checkstyle/checkstyle/pull/15311 also and we use both checks (whitespaceAfter and around) in ITs input. That's why I make a separate PR to extend input after merging both checks updates

Reply:
I am good with the separate PR, let's get this across the finish line